dags/common/request.py

```diff
@@ -0,0 +1,53 @@
+from urllib import response
+from requests.exceptions import RequestException
+
+class Request:
+    def __init__(
+        self,
+        base_url: str = 'localhost',
+        port: int = 8080,
+        parameters: dict = {},
+        subdir: str ='',
+        headers: dict = {}
+    ):
+        self.base_url = base_url 
+        self.parameters = parameters
+        self.headers = headers
+        self.subdir = subdir
+
+    def __repr__(self) -> str:
+        return(f"< RestApi base_url={self.base_url}, parameters={self.parameters}, header={self.headers} >")
+        
+    def get_response(self):
+        from furl import furl
+        import requests
+
+        url_base_obj = furl(self.base_url)
+        if bool(self.subdir):
+            url_base_obj = url_base_obj.path(self.subdir)
+        if bool(self.parameters):
+            url_base_obj = url_base_obj.add(self.parameters)
+        if bool(self.headers):
+            response = requests.get(url_base_obj.url, headers=self.headers)
+            return response
+        response=requests.get(url_base_obj.url)
+        return response
+
+    def get_response_content(self):
+        response = self.get_response()
+        if response.status_code == 200:
+            return response.content
+        print(f'Error while trying to make a request, status code {response.status_code}')
+        raise RequestException
+
+    def set(self, base_url="", parameters={}, headers={}):
```

Comment:
This should be removed. If we need to change something it means we're running a new request => new instance of the object